pkg/distributor/push.go

```diff
@@ -141,12 +147,42 @@ func handler(
 			if code != 202 {
 				level.Error(logger).Log("msg", "push error", "err", err)
 			}
-			addHeaders(w, err)
+			addHeaders(w, err, r, code, retryCfg, logger)
 			http.Error(w, msg, code)
 		}
 	})
 }
 
+/*
+calculateRetryAfter derives the value for the "Retry-After" header based on the provided HTTP request and RetryConfig.
+
+It calculates the delay in seconds for retry attempts, considering factors such as the Retry-Attempt header,
+RetryConfig parameters, and exponential backoff strategy. If Retry-Attempt is not valid or less than 1, it defaults to 1.
+If Retry-Attempt exceeds MaxAllowedAttempts from RetryConfig, it is capped at MaxAllowedAttempts.
+
+The delaySeconds is then calculated as a random number within the range of exponential backoff between minRetry and maxRetry.
+
+Returns: A string representing the calculated value for the "Retry-After" header.
+*/
+func calculateRetryAfter(req *http.Request, retryCfg *RetryConfig) string {
```

Comment:
nitpick, but can we just accept the value of `Retry-Attempt` as a string instead of the whole request? it keeps the scope of the function smaller